src/app/features/vendor/vendor-creation/vendor-creation.component.ts

```diff
@@ -1,9 +1,127 @@
 import { Component } from '@angular/core';
+import { IVendor } from '../models/vendor.model';
+import { FormBuilder, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
+
+// LLD
+
+// This component is to add new vendors
+
+// it has the fields:
+
+// Profile Info:
+// Vendor Name: Required text input.
+// State: Required input Dropdown with state options.
+// Country: Required input Dropdown with country options.
+// Markets: Required input Multi-select dropdown with  options.
+// Contact Info:
+// Email: Required email input.
+// Phone: Required Input with country code dropdown.
+// Website: Optional text input.
+// Service Category:
+// Service: Required input Dropdown with  options.
+
+// It has the buttons:
+// Submit for approval
+// Cancel
+
+// This screen contains following actions:
+// Input values to different fields.
+// Submitting the data and going to View Profile page.
+// Cancelling the changes made and going back to Vendor Listing Page.
+
+// Submit:
+// When the user clicks on the submit for approval button a modal wil appear asking the confirmation  for data submission,
+// if user confirms, user will be directed to view profile page.
+
+// Cancel:
+// When the user clicks on cancel button a modal will appear asking confirmation for cancelling the data entered without submitting,
+// if user confirms , user will be directed back to Vendor Listing page.
 
 @Component({
   selector: 'app-vendor-creation',
   standalone: false,
   templateUrl: './vendor-creation.component.html',
   styleUrl: './vendor-creation.component.scss',
 })
-export class VendorCreationComponent {}
+
+export class VendorCreationComponent {
+
+addVendorForm!: FormGroup;
+vendor!: IVendor[] ;
+
+constructor(private readonly fb: FormBuilder) {}
+
+ngOnInit(){
+  this.vendor = [
```

Comment:
Is this vendor list using in this component?

Reply:
yes just to fetch options onto dropdown, its just dummy

Reply:
Have changed to only required data